pdr_backend/lake/persistent_data_store.py

```diff
@@ -196,15 +258,45 @@ def move_table_data(self, temp_table, permanent_table_name: str):
 
         # check if the permanent table exists
         if permanent_table_name not in table_names:
+            # select one row from the temporary table
+            temp_table_df = self.duckdb_conn.execute(
+                f"SELECT * FROM {temp_table.fullname} LIMIT 1"
+            ).pl()
+
+            # create the sql schema
+            sql_table_schema = self._create_sql_table_schema(temp_table_df.schema)
+
+            # create the permanent table
+            self.execute_sql(
+                f"CREATE TABLE {permanent_table_name} ({sql_table_schema})"
+            )
```

Comment:
can we put all of this into a function?

Reply:
https://github.com/oceanprotocol/pdr-backend/pull/993#discussion_r1593838809